rfc/updates.md

```diff
@@ -0,0 +1,199 @@
+* After the agent returned successfully, update the Kubernetes version to `1.23.12` and components reference to `k8s-components-1.23.12` in the `k8s-version` ConfigMap 
+* Now, iterate over all nodes, and replace them if their Kubernetes version is outdated
+
+
+## Extending the `constellation upgrade` command
+
+Currently, `constellation upgrade` allows us to upgrade the VM image via the following entry in the constellation-config.yaml:
+
+```yaml
+upgrade:
+  image: /communityGalleries/ConstellationCVM-b3782fa0-0df7-4f2f-963e-fc7fc42663df/images/constellation/versions/2.3.0
+  measurements:
+    11: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
+    12: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
+```
+
+Instead of having a separate `upgrade` section, we will opt for a declarative approach by updating the existing values of the config file. 
+
+```yaml
+kubernetesVersion: 1.24.3
+kubernetesServicesVersion: 1.24.5 # Bundled Kubernetes components (Autoscaler, CloudControllerManager, CloudNodeManager, GCP Guest Agent, Konnectivity) 
+constellationVersion: 2.2.0 # or microserviceVersion: (KMS, AccessManager, JoinService, NodeMaintainanceOperator, NodeOperator, OLM, Verification, Cilium)
+provider:
+  azure:
+    image: /communityGalleries/ConstellationCVM-b3782fa0-0df7-4f2f-963e-fc7fc42663df/images/constellation/versions/2.3.0
+    measurements:
+    11: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
+    12: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
+```
+
+Note that:
+* `constellationVersion` 2.2.0 contains components which are all released in version 2.2.0
+* `kubernetesServicesVersion` 1.24.5 could contain Autoscaler 1.24.2, CCM 1.24.8 since their patch versions are not in sync with Kubernetes. Moreover, those component versions will be bundled by us. Think: public lookup table from `kubernetesServicesVersion` -> component version. 
+
+When `constellation upgrade execute` is called the CLI needs to perform the following steps:
+
+1. create a new `k8s-components-1.24.3` ConfigMap with the corresponding URLs and hashes from the lookup table in the CLI
+2. update the measurements in the `join-config` ConfigMap
+3. update the Kubernetes version and VM image in the `nodeimage` CRD
+4. update Cilium + Constellation microservices
+
+The actual update in step 2. and 3. will be handled by the node-operator inside Constellation. Step 4. will be done via client side helm deployments.
+
+Since the actual Kubernetes components and Constellation microservice versions are hidden, we will show the user for the actual changes taking place:
+
+```bash
+$ constellation upgrade execute
+Updating wanted Kubernetes version to 1.24.3 ...
+Updating wanted Image to /communityGalleries/ConstellationCVM-b3782fa0-0df7-4f2f-963e-fc7fc42663df/images/constellation/versions/2.3.0
+
+Updating Kubernetes services version to 1.24.5:
+  Autoscaler: 1.24.3 --> 1.24.3
```

Comment:
Should we add an additional note, e.g., `(not updated)`, to clearly show the user that keeping the same version is correct and not a typo/bug?